sdk/aqualink_sdk/api/aggregate.py

```diff
@@ -0,0 +1,49 @@
+import pandas as pd
+from pprint import pprint
+
+default_mapping = {
+    'min_temperature':'min',
+    'precipitation':'sum',
+    'alert_level':'max',
+    'bottom_temperature':'mean'
+}
+
+# TODO - Build on top of TimeSeriesApi ?
+
+def aggregate_data(timeseries_data, aggregate_frequency, aggregate_mapping=default_mapping):
+    # TODO - verify that all the metrics requested in "metrics" have a mapping
+    # TODO - strongly type the frequency or regex
+    # TODO - Hourly and aggregate are incompatible
+
+    data = timeseries_data._data_store
+    # Data is in the form:
+    # {'bottom_temperature': {
+    #     'spotter': {
+    #         'data': [
+    #             {'timestamp': '2022-01-08T00:04:00.000Z', 'value': 25.32},
+    #             {'timestamp': '2022-01-08T00:14:00.000Z', 'value': 25.259999999999998},
+    #             {'timestamp': '2022-01-08T00:24:00.000Z', 'value': 25.28}]}}
+    # }
+
+    new_data = {}
+    for metric_name, metric_data in data.items():
+        for sonde_type_name, sonde_data in metric_data.items():
+            dataframe = pd.DataFrame(sonde_data['data'])
```

Comment:
Depending on how big the input data is, it might help to tell pandas what data types you have in columns. Speeds up conversion to DF, and might provide a bit of validation.